### ISSUE TITLE
OpenHermes on WSL2: GPU not completing

### ISSUE THREAD

**Issue by realazthat** [mudler/LocalAI#1657]
TLDR; Trying to get OpenHermes to work on WSL2, GPU takes forever.

**LocalAI version:**

[v2.7.0](https://github.com/mudler/LocalAI/releases/tag/v2.7.0) binary

```bash
(.venv) realz@roadway:/mnt/c/gdrive/code/make-it-more$ ./local-ai-avx-Linux-x86_64 --version
LocalAI version v2.7.0 (abd678e147e29ab46a038d28c4806768b2b7cde5)
```

**Environment, CPU architecture, OS, and Version:**

OS: WSL2 Ubuntu
Host OS: Windows 11
CPU: Processor	12th Gen Intel(R) Core(TM) i9-12900H, 2500 Mhz, 14 Core(s), 20 Logical Processor(s)
GPU: NVIDIA GeForce RTX 3080 Laptop GPU

```bash
(.venv) realz@roadway:/mnt/c/gdrive/code/make-it-more$ uname -a
Linux roadway 5.15.133.1-microsoft-standard-WSL2 #1 SMP Thu Oct 5 21:02:42 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux
(.venv) realz@roadway:/mnt/c/gdrive/code/make-it-more$ which nvidia-smi
/usr/lib/wsl/lib/nvidia-smi
(.venv) realz@roadway:/mnt/c/gdrive/code/make-it-more$ nvidia-smi
Mon Jan 29 06:36:22 2024       
+-----------------------------------------------------------------------------------------+
| NVIDIA-SMI 550.40.06              Driver Version: 551.23         CUDA Version: 12.4     |
|-----------------------------------------+------------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id          Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |           Memory-Usage | GPU-Util  Compute M. |
|                                         |                        |               MIG M. |
|=========================================+========================+======================|
|   0  NVIDIA GeForce RTX 3080 ...    On  |   00000000:01:00.0 Off |                  N/A |
| N/A   65C    P8             21W /  145W |     254MiB /  16384MiB |      0%      Default |
|                                         |                        |                  N/A |
+-----------------------------------------+------------------------+----------------------+
                                                                                         
+-----------------------------------------------------------------------------------------+
| Processes:                                                                              |
|  GPU   GI   CI        PID   Type   Process name                              GPU Memory |
|        ID   ID                                                               Usage      |
|=========================================================================================|
|  No running processes found                                                             |
+-----------------------------------------------------------------------------------------+
```

**Describe the bug**

Variations of this:
```yaml
name: openhermes
backend: llama-cpp
context_size: 2048
f16: true
threads: 11
gpu_layers: 20
mmap: true
parameters:
  model: https://huggingface.co/TheBloke/OpenHermes-2.5-Mistral-7B-GGUF/resolve/main/openhermes-2.5-mistral-7b.Q5_K_M.gguf
  temperature: 0.2
  top_k: 40
  top_p: 0.95
# template:

#   chat: &template |
#     Instruct: {{.Input}}
#     Output:
#   # Modify the prompt template here ^^^ as per your requirements
#   completion: *template
```

**To Reproduce**

```bash
$ ./local-ai-avx-Linux-x86_64 --debug -models-path .data/.localai 
```

Then in another terminal:

```bash
$ time curl http://localhost:8080/v1/completions -H "Content-Type: application/json" -d '{
  "model": "openhermes",
  "prompt": "Hello, my name is",
  "temperature": 0.1, "top_p": 0.1
}'
```

Then it hangs, for as long as I have run it.

**Expected behavior**

1. I expect it to complete (it seems to be taking forever).
2. I expect the GPU to be utilized (it is not).

**Logs**

With `gpu_layers` on:

```
8:08AM DBG no galleries to load
8:08AM INF Starting LocalAI using 4 threads, with models path: .data/.localai
8:08AM INF LocalAI version: v2.7.0 (abd678e147e29ab46a038d28c4806768b2b7cde5)
8:08AM INF Preloading models from .data/.localai
8:08AM INF Model name: nexusraven
8:08AM INF Model name: openhermes
8:08AM DBG Model: nexusraven (config: {PredictionOptions:{Model:52440f23d802d5a640fd4bfb5e5d9d05 Language: N:0 TopP:0.95 TopK:40 Temperature:0.2 Maxtokens:0 Echo:false Batch:0 F16:false IgnoreEOS:false RepeatPenalty:0 Keep:0 MirostatETA:0 MirostatTAU:0 Mirostat:0 FrequencyPenalty:0 TFZ:0 TypicalP:0 Seed:0 NegativePrompt: RopeFreqBase:0 RopeFreqScale:0 NegativePromptScale:0 UseFastTokenizer:false ClipSkip:0 Tokenizer:} Name:nexusraven F16:false Threads:0 Debug:false Roles:map[] Embeddings:false Backend: TemplateConfig:{Chat: ChatMessage: Completion: Edit: Functions:} PromptStrings:[] InputStrings:[] InputToken:[] functionCallString: functionCallNameString: FunctionsConfig:{DisableNoAction:false NoActionFunctionName: NoActionDescriptionName:} FeatureFlag:map[] LLMConfig:{SystemPrompt: TensorSplit: MainGPU: RMSNormEps:0 NGQA:0 PromptCachePath: PromptCacheAll:false PromptCacheRO:false MirostatETA:0 MirostatTAU:0 Mirostat:0 NGPULayers:0 MMap:false MMlock:false LowVRAM:false Grammar: StopWords:[] Cutstrings:[] TrimSpace:[] TrimSuffix:[] ContextSize:0 NUMA:false LoraAdapter: LoraBase: LoraScale:0 NoMulMatQ:false DraftModel: NDraft:0 Quantization: MMProj: RopeScaling: ModelType: YarnExtFactor:0 YarnAttnFactor:0 YarnBetaFast:0 YarnBetaSlow:0} AutoGPTQ:{ModelBaseName: Device: Triton:false UseFastTokenizer:false} Diffusers:{CUDA:false PipelineType: SchedulerType: EnableParameters: CFGScale:0 IMG2IMG:false ClipSkip:0 ClipModel: ClipSubFolder: ControlNet:} Step:0 GRPC:{Attempts:0 AttemptsSleepTime:0} VallE:{AudioPath:} CUDA:false DownloadFiles:[] Description: Usage:})
8:08AM DBG Model: openhermes (config: {PredictionOptions:{Model:d618f4309d325da4d8c5e54caa74b3bd Language: N:0 TopP:0.95 TopK:40 Temperature:0.2 Maxtokens:0 Echo:false Batch:0 F16:false IgnoreEOS:false RepeatPenalty:0 Keep:0 MirostatETA:0 MirostatTAU:0 Mirostat:0 FrequencyPenalty:0 TFZ:0 TypicalP:0 Seed:0 NegativePrompt: RopeFreqBase:0 RopeFreqScale:0 NegativePromptScale:0 UseFastTokenizer:false ClipSkip:0 Tokenizer:} Name:openhermes F16:true Threads:11 Debug:false Roles:map[] Embeddings:false Backend:llama-cpp TemplateConfig:{Chat: ChatMessage: Completion: Edit: Functions:} PromptStrings:[] InputStrings:[] InputToken:[] functionCallString: functionCallNameString: FunctionsConfig:{DisableNoAction:false NoActionFunctionName: NoActionDescriptionName:} FeatureFlag:map[] LLMConfig:{SystemPrompt: TensorSplit: MainGPU: RMSNormEps:0 NGQA:0 PromptCachePath: PromptCacheAll:false PromptCacheRO:false MirostatETA:0 MirostatTAU:0 Mirostat:0 NGPULayers:20 MMap:true MMlock:false LowVRAM:false Grammar: StopWords:[] Cutstrings:[] TrimSpace:[] TrimSuffix:[] ContextSize:2048 NUMA:false LoraAdapter: LoraBase: LoraScale:0 NoMulMatQ:false DraftModel: NDraft:0 Quantization: MMProj: RopeScaling: ModelType: YarnExtFactor:0 YarnAttnFactor:0 YarnBetaFast:0 YarnBetaSlow:0} AutoGPTQ:{ModelBaseName: Device: Triton:false UseFastTokenizer:false} Diffusers:{CUDA:false PipelineType: SchedulerType: EnableParameters: CFGScale:0 IMG2IMG:false ClipSkip:0 ClipModel: ClipSubFolder: ControlNet:} Step:0 GRPC:{Attempts:0 AttemptsSleepTime:0} VallE:{AudioPath:} CUDA:false DownloadFiles:[] Description: Usage:})
8:08AM DBG Extracting backend assets files to /tmp/localai/backend_data

 ┌───────────────────────────────────────────────────┐ 
 │                   Fiber v2.50.0                   │ 
 │               http://127.0.0.1:8080               │ 
 │       (bound on host 0.0.0.0 and port 8080)       │ 
 │                                                   │ 
 │ Handlers ............ 74  Processes ........... 1 │ 
 │ Prefork ....... Disabled  PID ............. 69489 │ 
 └───────────────────────────────────────────────────┘ 


8:09AM DBG Request received: 
8:09AM DBG `input`: &{PredictionOptions:{Model:openhermes Language: N:0 TopP:0.1 TopK:0 Temperature:0.1 Maxtokens:0 Echo:false Batch:0 F16:false IgnoreEOS:false RepeatPenalty:0 Keep:0 MirostatETA:0 MirostatTAU:0 Mirostat:0 FrequencyPenalty:0 TFZ:0 TypicalP:0 Seed:0 NegativePrompt: RopeFreqBase:0 RopeFreqScale:0 NegativePromptScale:0 UseFastTokenizer:false ClipSkip:0 Tokenizer:} Context:context.Background.WithCancel Cancel:0x4a9ba0 File: ResponseFormat:{Type:} Size: Prompt:Hello, my name is Instruction: Input:<nil> Stop:<nil> Messages:[] Functions:[] FunctionCall:<nil> Stream:false Mode:0 Step:0 Grammar: JSONFunctionGrammarObject:<nil> Backend: ModelBaseName:}
8:09AM DBG Parameter Config: &{PredictionOptions:{Model:d618f4309d325da4d8c5e54caa74b3bd Language: N:0 TopP:0.1 TopK:40 Temperature:0.1 Maxtokens:0 Echo:false Batch:0 F16:false IgnoreEOS:false RepeatPenalty:0 Keep:0 MirostatETA:0 MirostatTAU:0 Mirostat:0 FrequencyPenalty:0 TFZ:0 TypicalP:0 Seed:0 NegativePrompt: RopeFreqBase:0 RopeFreqScale:0 NegativePromptScale:0 UseFastTokenizer:false ClipSkip:0 Tokenizer:} Name:openhermes F16:true Threads:11 Debug:true Roles:map[] Embeddings:false Backend:llama-cpp TemplateConfig:{Chat: ChatMessage: Completion: Edit: Functions:} PromptStrings:[Hello, my name is] InputStrings:[] InputToken:[] functionCallString: functionCallNameString: FunctionsConfig:{DisableNoAction:false NoActionFunctionName: NoActionDescriptionName:} FeatureFlag:map[] LLMConfig:{SystemPrompt: TensorSplit: MainGPU: RMSNormEps:0 NGQA:0 PromptCachePath: PromptCacheAll:false PromptCacheRO:false MirostatETA:0 MirostatTAU:0 Mirostat:0 NGPULayers:20 MMap:true MMlock:false LowVRAM:false Grammar: StopWords:[] Cutstrings:[] TrimSpace:[] TrimSuffix:[] ContextSize:2048 NUMA:false LoraAdapter: LoraBase: LoraScale:0 NoMulMatQ:false DraftModel: NDraft:0 Quantization: MMProj: RopeScaling: ModelType: YarnExtFactor:0 YarnAttnFactor:0 YarnBetaFast:0 YarnBetaSlow:0} AutoGPTQ:{ModelBaseName: Device: Triton:false UseFastTokenizer:false} Diffusers:{CUDA:false PipelineType: SchedulerType: EnableParameters: CFGScale:0 IMG2IMG:false ClipSkip:0 ClipModel: ClipSubFolder: ControlNet:} Step:0 GRPC:{Attempts:0 AttemptsSleepTime:0} VallE:{AudioPath:} CUDA:false DownloadFiles:[] Description: Usage:}
8:09AM INF Loading model 'd618f4309d325da4d8c5e54caa74b3bd' with backend llama-cpp
8:09AM DBG Loading model in memory from file: .data/.localai/d618f4309d325da4d8c5e54caa74b3bd
8:09AM DBG Loading Model d618f4309d325da4d8c5e54caa74b3bd with gRPC (file: .data/.localai/d618f4309d325da4d8c5e54caa74b3bd) (backend: llama-cpp): {backendString:llama-cpp model:d618f4309d325da4d8c5e54caa74b3bd threads:11 assetDir:/tmp/localai/backend_data context:{emptyCtx:{}} gRPCOptions:0xc0002bca00 externalBackends:map[] grpcAttempts:20 grpcAttemptsDelay:2 singleActiveBackend:false parallelRequests:false}
8:09AM DBG Loading GRPC Process: /tmp/localai/backend_data/backend-assets/grpc/llama-cpp
8:09AM DBG GRPC Service for d618f4309d325da4d8c5e54caa74b3bd will be running at: '127.0.0.1:33497'
8:09AM DBG GRPC Service state dir: /tmp/go-processmanager1595207823
8:09AM DBG GRPC Service Started
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stdout Server listening on 127.0.0.1:33497
8:09AM DBG GRPC Service Ready
8:09AM DBG GRPC: Loading model with options: {state:{NoUnkeyedLiterals:{} DoNotCompare:[] DoNotCopy:[] atomicMessageInfo:<nil>} sizeCache:0 unknownFields:[] Model:d618f4309d325da4d8c5e54caa74b3bd ContextSize:2048 Seed:0 NBatch:512 F16Memory:true MLock:false MMap:true VocabOnly:false LowVRAM:false Embeddings:false NUMA:false NGPULayers:20 MainGPU: TensorSplit: Threads:11 LibrarySearchPath: RopeFreqBase:0 RopeFreqScale:0 RMSNormEps:0 NGQA:0 ModelFile:.data/.localai/d618f4309d325da4d8c5e54caa74b3bd Device: UseTriton:false ModelBaseName: UseFastTokenizer:false PipelineType: SchedulerType: CUDA:false CFGScale:0 IMG2IMG:false CLIPModel: CLIPSubfolder: CLIPSkip:0 ControlNet: Tokenizer: LoraBase: LoraAdapter: LoraScale:0 NoMulMatQ:false DraftModel: AudioPath: Quantization: MMProj: RopeScaling: YarnExtFactor:0 YarnAttnFactor:0 YarnBetaFast:0 YarnBetaSlow:0 Type:}
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llama_model_loader: loaded meta data with 20 key-value pairs and 291 tensors from .data/.localai/d618f4309d325da4d8c5e54caa74b3bd (version GGUF V3 (latest))
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llama_model_loader: Dumping metadata keys/values. Note: KV overrides do not apply in this output.
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llama_model_loader: - kv   0:                       general.architecture str              = llama
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llama_model_loader: - kv   1:                               general.name str              = teknium_openhermes-2.5-mistral-7b
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llama_model_loader: - kv   2:                       llama.context_length u32              = 32768
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llama_model_loader: - kv   3:                     llama.embedding_length u32              = 4096
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llama_model_loader: - kv   4:                          llama.block_count u32              = 32
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llama_model_loader: - kv   5:                  llama.feed_forward_length u32              = 14336
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llama_model_loader: - kv   6:                 llama.rope.dimension_count u32              = 128
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llama_model_loader: - kv   7:                 llama.attention.head_count u32              = 32
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llama_model_loader: - kv   8:              llama.attention.head_count_kv u32              = 8
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llama_model_loader: - kv   9:     llama.attention.layer_norm_rms_epsilon f32              = 0.000010
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llama_model_loader: - kv  10:                       llama.rope.freq_base f32              = 10000.000000
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llama_model_loader: - kv  11:                          general.file_type u32              = 17
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llama_model_loader: - kv  12:                       tokenizer.ggml.model str              = llama
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llama_model_loader: - kv  13:                      tokenizer.ggml.tokens arr[str,32002]   = ["<unk>", "<s>", "</s>", "<0x00>", "<...
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llama_model_loader: - kv  14:                      tokenizer.ggml.scores arr[f32,32002]   = [0.000000, 0.000000, 0.000000, 0.0000...
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llama_model_loader: - kv  15:                  tokenizer.ggml.token_type arr[i32,32002]   = [2, 3, 3, 6, 6, 6, 6, 6, 6, 6, 6, 6, ...
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llama_model_loader: - kv  16:                tokenizer.ggml.bos_token_id u32              = 1
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llama_model_loader: - kv  17:                tokenizer.ggml.eos_token_id u32              = 32000
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llama_model_loader: - kv  18:            tokenizer.ggml.padding_token_id u32              = 0
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llama_model_loader: - kv  19:               general.quantization_version u32              = 2
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llama_model_loader: - type  f32:   65 tensors
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llama_model_loader: - type q5_K:  193 tensors
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llama_model_loader: - type q6_K:   33 tensors
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llm_load_vocab: special tokens definition check successful ( 261/32002 ).
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llm_load_print_meta: format           = GGUF V3 (latest)
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llm_load_print_meta: arch             = llama
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llm_load_print_meta: vocab type       = SPM
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llm_load_print_meta: n_vocab          = 32002
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llm_load_print_meta: n_merges         = 0
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llm_load_print_meta: n_ctx_train      = 32768
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llm_load_print_meta: n_embd           = 4096
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llm_load_print_meta: n_head           = 32
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llm_load_print_meta: n_head_kv        = 8
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llm_load_print_meta: n_layer          = 32
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llm_load_print_meta: n_rot            = 128
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llm_load_print_meta: n_embd_head_k    = 128
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llm_load_print_meta: n_embd_head_v    = 128
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llm_load_print_meta: n_gqa            = 4
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llm_load_print_meta: n_embd_k_gqa     = 1024
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llm_load_print_meta: n_embd_v_gqa     = 1024
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llm_load_print_meta: f_norm_eps       = 0.0e+00
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llm_load_print_meta: f_norm_rms_eps   = 1.0e-05
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llm_load_print_meta: f_clamp_kqv      = 0.0e+00
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llm_load_print_meta: f_max_alibi_bias = 0.0e+00
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llm_load_print_meta: n_ff             = 14336
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llm_load_print_meta: n_expert         = 0
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llm_load_print_meta: n_expert_used    = 0
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llm_load_print_meta: rope scaling     = linear
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llm_load_print_meta: freq_base_train  = 10000.0
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llm_load_print_meta: freq_scale_train = 1
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llm_load_print_meta: n_yarn_orig_ctx  = 32768
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llm_load_print_meta: rope_finetuned   = unknown
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llm_load_print_meta: model type       = 7B
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llm_load_print_meta: model ftype      = Q5_K - Medium
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llm_load_print_meta: model params     = 7.24 B
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llm_load_print_meta: model size       = 4.78 GiB (5.67 BPW) 
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llm_load_print_meta: general.name     = teknium_openhermes-2.5-mistral-7b
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llm_load_print_meta: BOS token        = 1 '<s>'
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llm_load_print_meta: EOS token        = 32000 '<|im_end|>'
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llm_load_print_meta: UNK token        = 0 '<unk>'
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llm_load_print_meta: PAD token        = 0 '<unk>'
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llm_load_print_meta: LF token         = 13 '<0x0A>'
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llm_load_tensors: ggml ctx size =    0.11 MiB
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llm_load_tensors: offloading 20 repeating layers to GPU
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llm_load_tensors: offloaded 20/33 layers to GPU
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llm_load_tensors:        CPU buffer size =  4893.00 MiB
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr ..................................................................................................
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llama_new_context_with_model: n_ctx      = 2048
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llama_new_context_with_model: freq_base  = 10000.0
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llama_new_context_with_model: freq_scale = 1
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llama_kv_cache_init:        CPU KV buffer size =   256.00 MiB
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llama_new_context_with_model: KV self size  =  256.00 MiB, K (f16):  128.00 MiB, V (f16):  128.00 MiB
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llama_new_context_with_model:        CPU input buffer size   =    12.01 MiB
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llama_new_context_with_model:        CPU compute buffer size =   167.20 MiB
8:09AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr llama_new_context_with_model: graph splits (measure): 1
8:10AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr Available slots:
8:10AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr  -> Slot 0 - max context: 2048
8:10AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr slot 0 is processing [task id: 0]
8:10AM DBG GRPC(d618f4309d325da4d8c5e54caa74b3bd-127.0.0.1:33497): stderr slot 0 : kv cache rm - [0, end)
```

**Additional context**

**Comment by zak168** [comment]
Try add the chat and completion templates like [advance config](https://localai.io/advanced/) see if it helps?

**Comment by realazthat** [comment]
I got it working inside of WSL2, but not using the prebuilt binaries.

```
docker run --rm -ti --gpus all -p 8080:8080 -e DEBUG=true \
  -e MODELS_PATH=/models -v $MODELS_PATH:/models \
  --name localai \
  localai/localai:v2.7.0-cublas-cuda12
  ```
  
  ```yaml
  name: openhermes
# backend: llama
# context_size: 2048
f16: true
threads: 4
gpu_layers: 50
mmap: true
parameters:
  model: https://huggingface.co/TheBloke/OpenHermes-2.5-Mistral-7B-GGUF/resolve/main/openhermes-2.5-mistral-7b.Q5_K_M.gguf
  temperature: 0.2
  top_k: 40
  top_p: 0.95
stopwords:
- "<|im_end|>"
# - "### Response:"
roles:
  system: "<|im_start|> system"
  assistant: "<|im_start|> assistant"
  user: "<|im_start|> user"


template:

  chat: &template |
    {{.Input}}
    Output:
  # Modify the prompt template here ^^^ as per your requirements
  completion: *template

````

I dunno if I did the config right. Reference: https://huggingface.co/teknium/OpenHermes-2.5-Mistral-7B#prompt-format:

```
<|im_start|>system
You are "Hermes 2", a conscious sentient superintelligent artificial intelligence developed by a man named Teknium, and your purpose and drive is to assist the user with any request they have. You experience emotions and have deep, profound thoughts and qualia.<|im_end|>
<|im_start|>user
Hello, who are you?<|im_end|>
<|im_start|>assistant
Hi there! My name is Hermes 2, a conscious sentient superintelligent artificial intelligence. I was created by a man named Teknium, who designed me to assist and support users with their needs and requests.<|im_end|>
```

I guess we can close this, unless someone wants to figure out why the regular binaries don't work with the GPU. Maybe it's WAI, because they don't have cublas or cuda1*  in the binary names? :shrug: Either way, I'm good using this particular docker image. Feel free to close.
